### DISCUSSION TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### DISCUSSION THREAD

**Discussion by bbe-dw** [srl-labs/containerlab#1036] [Q&A] [answered]
Dear,

I am trying to set up a network test environment via containerlab. (We use Cumulus Linux in production)
After installing the prerequisites, I created a simple topology file.

name: DWlab1

topology:
  nodes:
    spine-test-01:
      kind: cvx
      image: networkop/cx:4.3.0
    leaf-test-01:
      kind: cvx
      image: networkop/cx:4.3.0
    leaf-test-02:
      kind: cvx
      image: networkop/cx:4.3.0

  links:
    - endpoints: ["spine-test-01:swp1", "leaf-test-01:swp51"]
    - endpoints: ["spine-test-01:swp2", "leaf-test-02:swp51"]
    - endpoints: ["leaf-test-01:swp49", "leaf-test-02:swp49"]

When I launch the containerlab deploy, I get this error message:

INFO[0000] Containerlab v0.32.0 started                 
INFO[0000] Parsing & checking topology file: DWlab1.clab.yml 
INFO[0000] Creating lab directory: /srv/lab/DWlab01/clab-DWlab1 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x68 pc=0x1ff0e26]

goroutine 40 [running]:
github.com/weaveworks/ignite/pkg/apis/ignite.(*VM).SetKernel(...)
	github.com/weaveworks/ignite@v0.10.0/pkg/apis/ignite/helpers.go:18
github.com/srl-labs/containerlab/runtime/ignite.(*IgniteRuntime).StartContainer(0xc00062aed0, {0x32c0a60, 0xc000b12300}, {0xc000617760?, 0x1c?}, 0xc00070ca80)
	github.com/srl-labs/containerlab/runtime/ignite/ignite.go:178 +0x2a6
github.com/srl-labs/containerlab/nodes/cvx.(*cvx).Deploy(0xc0005caec0, {0x32c0a60, 0xc000b12300})
	github.com/srl-labs/containerlab/nodes/cvx/cvx.go:82 +0x76
github.com/srl-labs/containerlab/clab.(*CLab).scheduleNodes.func1(0x580525?, 0xc00057e120, 0xc000502fd0?, {0x32c5b30, 0xc000092f90})
	github.com/srl-labs/containerlab/clab/clab.go:308 +0x3e7
created by github.com/srl-labs/containerlab/clab.(*CLab).scheduleNodes
	github.com/srl-labs/containerlab/clab/clab.go:339 +0x145
	

Have you ever encountered the case with cx images ? Could I have missed something?
Thanks for your help

**Comment by hellt** [comment]
Hi

Try to Set the runtime to docker with

name: ...
topology:
  default:
    runtime: docker


Also maybe @networkop knows if something has changed in ignite

**Reply by bbe-dw** [reply]
I would like to use the ignite method because container runtime method not supported MLAG.
Thanks for the suggestion

**Comment by networkop** [comment]
AFAIR, all ignite dependencies should be baked into containerlab and statically pinned. I'll test locally to confirm

**Reply by hellt** [reply]
It might be that I upgraded ignite in one of the housekeeping PRs. 
@bbe-dw you may want to try to downgrade to some old version like 0.27 to see if that is indeed a dependency issue 
In the installation instructions you will find a command to install a specific version

**Reply by networkop** [reply]
yep, can confirm it works with 0.27 but doesn't work with the latest version. what was the change you made @hellt ?

**Reply by hellt** [reply]
The only changes made were with ignite version. I updated it I think

**Reply by steiler** [reply]
`git diff v0.27.1 main go.mod`
doesn't seem like there has been an update on any weaveworks dependencies...

code changes to ignite.go also seem very limited
`git diff v0.27.1 main runtime/ignite/ignite.go`

**Reply by networkop** [reply]
seems like there were quite a lot of changes. one missing function 
https://github.com/srl-labs/containerlab/compare/v0.32.0..v0.27.0

**Reply by hellt** [reply]
I'd check if 0.31 works
one thing that has been changed in 0.32 that could cause that is that there is no more sequential dependency that was refactored by @steiler 
these inter-dependencies were introduced in 0.32, so if 0.31 works, then this is the culprit

**Reply by steiler** [reply]
The diff is the wrong way around... that is a function thats been added in v0.32.0 for the DependencyManagement.
All the rest seems to be just formatting.

**Reply by steiler** [reply]
right @hellt ... I remember now, I've mentioned the specific of what I might have broke in the PR.
https://github.com/srl-labs/containerlab/pull/976#issuecomment-1213220475

**Reply by hellt** [reply]
@steiler would you be able to have a look at that?

**Reply by steiler** [reply]
Should be this snippet thats no longer existing.
https://github.com/srl-labs/containerlab/blob/v0.31.1/cmd/deploy.go#L193-L195

so let me implement a additional `createIgniteDependency(...)` function. that results in dependency chain ... 

x -> nil
y -> x
z -> y 

would that be sufficient? @networkop

**Reply by networkop** [reply]
I don't fully understand your explanation. I do remember that ignite needs to create VMs sequentially, so as long as you can make that happen, I think it might work.

**Comment by hellt** [comment, marked as answer]
So as it was discovered in #1039 there are issues with ignite that appear after we upgraded some packages.
Unfortuntely we can't revert the package upgrade, so for now the ignite is out of support. cvx uses have two options

- use 0.27.1 containerlab
- or use docker runtime

**Comment by hellt** [comment]
Hi

Try to Set the runtime to docker with

name: ...
topology:
  default:
    runtime: docker
